### PR TITLE
populate the search bar and run a search query automatically

### PR DESCRIPTION
**Pull request by voltraeth**
## What does this PR do and why?

This PR populates the search bar with the related search query and runs a search query automatically.
https://github.com/jbx-protocol/juice-interface/issues/1566

## Screenshots or screen recordings
Screen Shot 2022-07-27 at 3 32 46 PM

https://user-images.githubusercontent.com/109549025/181384037-1023d57f-5113-4db2-8b5a-19579ea3782b.mov

## Acceptance checklist

- [ ] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [ ] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).
- [ ] I have tested this PR in dark mode and light mode (if applicable).

### CODE REVIEW

**Comment by tomquirk** [comment]
@voltraeth woohoo thanks for this contribution!

@peripheralist could you please review?

**Comment by tomquirk** [comment]
yo my bad, just noticed this is still in draft. don't mind me! 🙏

**Comment by voltraeth** [comment]
> yo my bad, just noticed this is still in draft. don't mind me! 🙏

just took it out of review! feel free to review and leave comments now

File: src/pages/projects/index.page.tsx [hunk 1]
```diff
@@ -58,11 +62,8 @@ export default function Projects() {
           return defaultTab
       }
     })
-  }, [userAddress, router.query])
-
-  const [searchText, setSearchText] = useState<string>(
-    (router.query.search as string | undefined) ?? '',
-  )
+    return setSearchText(router.query.search as string)
```
Comment (peripheralist) [inline src/pages/projects/index.page.tsx:65]: maybe this can just be called instead of returned?

**Review by peripheralist — COMMENTED** [review]
beautifully simple fix! thanks @voltraeth 

just a couple nitty react questions 🙏

File: src/pages/projects/index.page.tsx (hunk 1 above)
Comment (voltraeth) [reply src/pages/projects/index.page.tsx:65]: I was getting a typescript error earlier without the return. let me remove it and see if i'm still getting it.

File: src/pages/projects/index.page.tsx [hunk 2]
```diff
@@ -58,11 +62,8 @@ export default function Projects() {
...
+  }, [userAddress, router.query, router.query.search])
```
[the remark this answers is not in the document]
Comment (voltraeth) [reply src/pages/projects/index.page.tsx:66]: I figured since we're also checking router.query.tab it might be necessary

**Review by peripheralist — APPROVED** [review]
thanks @voltraeth , sorry for the delay on this!

**Comment by peripheralist** [comment]
approving this, but will wait to merge until some current urgent bug fixes are merged to avoid any conflicts (cc @tomquirk )

**Pull request merged**